driver/modern_bpf/helpers/extract/extract_from_kernel.h

```diff
@@ -380,21 +386,20 @@ static __always_inline pid_t extract__task_xid_vnr(struct task_struct *task, enu
  * @param type pid type.
  * @return `start_time` of init task struct from pid namespace seen from current task pid namespace.
  */
-static __always_inline u64 extract__task_pidns_start_time(struct task_struct *task, enum pid_type type)
+static __always_inline u64 extract__task_pidns_start_time(struct task_struct *task, enum pid_type type, long ret)
 {
-	struct pid *pid_struct = extract__task_pid_struct(task, type);
-	struct pid_namespace *pid_namespace = extract__namespace_of_pid(pid_struct);
-	struct task_struct *child_reaper = NULL;
-	u64 pidns_start_time = 0;
-	if(pid_namespace)
+	// only perform lookup when clone/vfork/fork returns 0 (child process / childtid)
+	if (ret == 0)
```

Comment:
Shouldn't this be `if (ret == 0 && pid_namespace)` like in driver and driver-bpf?